persistence.js

```diff
@@ -341,48 +318,38 @@ RedisPersistence.prototype._setup = function () {
   })
 }
 
-function processKeysForClient (all, that) {
-  var keys = Object.keys(all)
-  var key = ''
-  var decoded = null
-  var raw = null
-
-  for (var i = 0; i < keys.length; i++) {
-    key = keys[i]
-    raw = that[key]
-    if (!raw) {
-      continue
-    }
-    decoded = msgpack.decode(raw)
-    decoded.clientId = key
-
-    that._matcher.add(decoded.topic, decoded)
+function processKeysForClient (clientId, clientHash, that) {
+  var topics = Object.keys(clientHash)
+  for (var i = 0; i < topics.length; i++) {
+    var topic = topics[i]
+    that._matcher.add(topic, {
+      clientId: clientId,
+      topic: topic,
+      qos: clientHash[topic]
+    })
   }
 }
 
 RedisPersistence.prototype.outgoingEnqueue = function (sub, packet, cb) {
   var listKey = 'outgoing:' + sub.clientId
   var key = listKey + ':' + packet.brokerId + ':' + packet.brokerCounter
 
-  var multi = this._db.multi()
-  multi.set(key, msgpack.encode(new Packet(packet)))
-  multi.rpush(listKey, key)
-
-  multi.exec(cb)
+  var pipeline = this._getPipeline()
+  pipeline.rpush(listKey, key)
```

Comment:
by SMEMBERS we can't control the offline message flow to the client.
lets see what @mcollina thinks about this?